Neo4j.Driver/Neo4j.Driver/Preview/Mapping/DictAsRecord.cs

```diff
@@ -27,20 +28,91 @@ internal class DictAsRecord : IRecord
 
     public DictAsRecord(object dict, IRecord record)
     {
-        _dict = dict switch
+        var readOnlyDictionary = dict switch
         {
-            IEntity entity => entity.Properties,
+            IEntity entity => entity.Properties ?? new Dictionary<string, object>(),
             IReadOnlyDictionary<string, object> dictionary => dictionary,
             _ => throw new InvalidOperationException($"Cannot create a DictAsRecord from a {dict.GetType().Name}")
         };
 
+        // this is only used by the default mapper so make it case insensitive
+        _dict = readOnlyDictionary.ToDictionary(
+            kvp => kvp.Key,
+            kvp => kvp.Value,
+            StringComparer.OrdinalIgnoreCase);
```

Comment:
This is ordinal but the other is CultureInvariantIgnoreCase, we should only use one for consistency.

Reply:
Changed to CultureInvariantIgnoreCase